### PR TITLE
🦺 Moving to the new Test suite (refactoring tests, Chapter MCCXXVIII)

### PR DESCRIPTION
**Pull request by kellertuer**
This continues #839 with the goal of solely using the new test suite and finding the last few code cov lines to cover them as well (at time of writing 6 lines left) – see remaining tasks in #870

## Manifold tests to move to the new test suite

* [x] determinant_one_matrices
* [x] stiefel (done in #858)
* [x] unitary_matrices

### CODE REVIEW

**Comment by kellertuer** [comment]
Over on the show/repr PR for Manopt ( https://github.com/JuliaManifolds/Manopt.jl/pull/569 ) I noticed that these super-PRs are not that nice to keep momentum, e.g. this one here is very slow because I focus on that one for now.

@mateuszbaran  So does it make sense to move this list into an Issue instead and I do single smaller PRs of 1-2 manifolds each? I think that is nicer to work on and also makes it easier, since one does not have to merge master here so often.

So for this one I just have to check where I lost a few lines, but with a new issue I would afterwards merge this.
These test merges also do not have to become new versions, since that is more like an internal restructure.

**Review by mateuszbaran — COMMENTED** [review]
I think a bit too much is deleted here.

File: test/manifolds-old/circle_old.jl [hunk 1]
```diff
@@ -1,282 +0,0 @@
...
-        rand!(rng, M, p)
-        @test is_point(M, p)
-        rand!(rng, M, X; vector_at = p)
-        @test is_vector(M, p, X)
-    end
-    @testset "Test sym_rem" begin
-        p = 4.0 # not a point
-        p = sym_rem(p) # modulo to a point
-        @test is_point(M, p)
-    end
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:182]: Implicitly yes, though via project and not directy.

File: test/manifolds-old/circle_old.jl [hunk 2]
```diff
@@ -1,282 +0,0 @@
...
-
-        angles = map(pp -> exp(pp * im), [-π / 2, 0.0, π])
-        @test mean(Mc, angles) ≈ exp(-π * im / 2)
-        @test mean(Mc, angles, [1.0, 1.0, 1.0]) ≈ exp(-π * im / 2)
-        @test_throws ErrorException mean(Mc, [-1.0 + 0im, 1.0 + 0im])
-        @test_throws ErrorException mean(Mc, [-1.0 + 0im, 1.0 + 0im], [1.0, 1.0])
-
-        # volume
-        @test manifold_volume(Mc) ≈ 2 * π
-        @test volume_density(Mc, 1.0 + 0.0im, 2im) == 1.0
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:227]: Yes, see the lines https://github.com/JuliaManifolds/Manifolds.jl/blob/9d9ad122debf72dbdc1eb1d0f403b122c6244471/test/manifolds/test_circle.jl#L15-L32, where both an expectation is set for each of them and they are in the list of functions.

File: test/manifolds-old/circle_old.jl [hunk 3]
```diff
@@ -1,282 +0,0 @@
...
-    end
-    types = [Complex{Float64}]
-
-    @testset "small and large distance tests" begin
-        p = -0.42681766710748265 + 0.9043377018818392im
-        q = -0.42681766710748226 + 0.9043377018818393im
-        @test isapprox(distance(Mc, p, q), 4.041272810440265e-16)
-        @test isapprox(distance(Mc, p, -q), 3.1415926535897927; atol = eps())
-    end
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:236]: Not exactly. The first one is done with both arguments equal – not sure why that test was here? For the second, I also did not see a reason why that test is necessary?

Overall I am trying to find a balance: Test everything that should be tested, boundary cases after calling the test suite but not “overtest” (i.e.: sure we could check that exp works as expected 20x, but maybe more than 2x is not necessary?)

File: test/manifolds-old/circle_old.jl [hunk 4]
```diff
@@ -1,282 +0,0 @@
...
-    @testset "Mixed array dimensions for exp and PT" begin
-        # this is an issue on Julia 1.6 but not later releases
-        M = Circle()
-        p = fill(0.0)
-        Manifolds.exp_fused!(M, p, p, [1.0], 2.0)
-        @test p ≈ fill(2.0)
-        parallel_transport_to!(M, p, p, [4.0], p)
-        @test p ≈ fill(4.0)
-    end
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:281]: ah probably not. I am also not sure I want to support all this mixing? I mean it is nice if it works, but if we always require that you can mix all point types with all point types, that is bit of an exponential growth in tests in the long run.

File: test/manifolds-old/determinant_one_matrices.jl [hunk 5]
```diff
@@ -1,68 +0,0 @@
...
-    @testset "rng test" begin
-        M = DeterminantOneMatrices(2)
-        pX = zeros(2, 2)
-        Manifolds._ensure_nonzero_rng_determinant!(
-            Random.default_rng(),
-            get_embedding(M),
-            pX,
-        )
-        @test abs(det(pX)) > 1.0e-8
-    end
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/determinant_one_matrices.jl:67]: Ah yes, we can re-add this.

File: test/manifolds-old/circle_old.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:182]: Ah no, I even kept literally exactly those lines: https://github.com/JuliaManifolds/Manifolds.jl/blob/9d9ad122debf72dbdc1eb1d0f403b122c6244471/test/manifolds/test_circle.jl#L158-L163

File: test/manifolds-old/circle_old.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:236]: Ah, and I just saw, those lines are still present

https://github.com/JuliaManifolds/Manifolds.jl/blob/9d9ad122debf72dbdc1eb1d0f403b122c6244471/test/manifolds/test_circle.jl#L164-L170

File: test/manifolds-old/circle_old.jl (hunk 4 above)
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:281]: Oh I just noticed, this is still rpesent

https://github.com/JuliaManifolds/Manifolds.jl/blob/9d9ad122debf72dbdc1eb1d0f403b122c6244471/test/manifolds/test_circle.jl#L171-L179

File: test/manifolds-old/circle_old.jl [hunk 6]
```diff
@@ -1,282 +0,0 @@
...
-            0.0,
-            1.0,
-            0.5,
-            2.0,
-            ManifoldDiff.βdifferential_shortest_geodesic_startpoint,
-        ) === 2.0
-
-        # volume
-        @test manifold_volume(M) ≈ 2 * π
-        @test volume_density(M, 0.0, 2.0) == 1.0
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply test/manifolds-old/circle_old.jl:117]: It is done in the test suite for the last 2 lines actually, but also still as a separate case, see

https://github.com/JuliaManifolds/Manifolds.jl/blob/9d9ad122debf72dbdc1eb1d0f403b122c6244471/test/manifolds/test_circle.jl#L191-L206

so we could maybe even remove the last 2 lines.

**Comment by kellertuer** [comment]
Oh, very nice, with the fix on Tucker (the only 2 lines missing on master) and the re-added test for the safeguard in utils, we actually reach 100% coverage on this PR.

**Comment by mateuszbaran** [comment]
I have re-added a few tests that I think are not fully covered yet by the new ones so that this can be merged.

**Review by mateuszbaran — APPROVED** [review]
